### ISSUE TITLE
Make formats.image_to_html use the rendition.img_tag to allow overriding markup in rich text editor

### ISSUE THREAD

**Issue by dibell** [wagtail/wagtail#4236]
I have overridden the wagtail image to add a caption field as documented in docs/advanced_topics/images/custom_image_model. However, the formats.py:image_to_html does not use the rendition.img_tag so the new field can not be used in rich text embedded images.

The change appears to be relatively small. I will submit two PRs - one for stable/1.13x and one for master

Many thanks!

**Comment by gasman** [comment]
Hi @dibell - thanks for the PRs!

Overriding `img_tag` to return something other than an `<img>` element is not something we'd encourage really - it's liable to break layouts within the admin backend. I'm also not sure that having a particular HTML presentation "baked in" to the rendition model is the right thing to do conceptually - I think it would be more appropriate to specify that presentation at the point of inserting it on the template, e.g. with a custom `{% captioned_image %}` tag that wraps the built-in `{% image %}` tag.

In the past we've implemented captioned images in rich text using a custom image format definition: https://github.com/torchbox/verdant-rca/blob/master/django-verdant/rca/image_formats.py - however, we found it was pushing the limits of what's feasible within rich text, and these days we'd go with StreamField for this kind of mixed content.

Having said that, I think this is a worthwhile change to make in the interests of reducing code duplication - that way, the option of overriding `img_tag` is available, but we wouldn't document it. The one minor nuisance is that changing `extra_attributes` from a string to a dict breaks backward compatibility with existing code that subclasses `Format`, like the snippet above - but I think that's acceptable given that these methods aren't officially documented.
